Add Recommendations for Using `astro deploy --image-name` (#1506)
astronomer/docs: astro/deploy-code.md

astro deploy --dags
```

> ## Deploy custom Docker images

Reviewer: ```suggestion
## Deploy a pre-built Docker image
```

I think it's more technically accurate to say pre-built? In some way, any Docker image you push is "custom" in the sense that it has your packages and code in it. This command is really just about pushing a Docker image that is pre-built

> You can run `astro deploy` to build your Astro project as a Docker image and deploy it, or you can run `astro deploy --image-name <custom-image>` to deploy a pre-built Astro Runtime image. To use this command, the image you deploy must be based on Astro Runtime and be available in a local Docker registry. 

Reviewer: ```suggestion
By default, running `astro deploy` with the Astro CLI both builds your Astro project into a Docker image and deploys it to Astro. To skip the build step and deploy a pre-built Docker image, run `astro deploy --image-name <image-name>` or specify the command in a CI/CD pipeline.

Deploying a pre-built Docker image can enable your team to:

- Test code changes across environments more consistently and with more control. You can build a Docker image once and promote the same Docker image across Deployments instead of rebuilding it each time.
- Lower the time it takes to deploy. If your Astro project has a number of packages that take a long time to install, it can be more efficient to build it separately.
- Securely specify additional mounts and arguments in your project, which is required for setups such as [installing Python packages from private sources](develop-project.md#install-python-packages-from-private-sources).

To use this command, the image you deploy must be based on Astro Runtime and be available in a local Docker registry. 
```

I'd rather just simplify the benefits here?

> Customizing your Runtime image lets you securely mount additional files and arguments in your project, which is required for setups such as [installing Python packages from private sources](develop-project.md#install-python-packages-from-private-sources).
> 
> You can also use the `--image-name` flag to separate your build and deploy steps in a CI/CD pipeline. This allows you more control over how your images are tested and deployed. 
> 

Reviewer: ```suggestion
```

You can use the `--image-name` flag with the following local development commands to test your image before deploying:

- `astro dev start`
- `astro dev restart`
- `astro dev parse`
- `astro dev pytest`

For more information about this command, see the [CLI command reference](cli/astro-deploy.md).

:::info

> If you're using an Apple M1 computer to deploy an Astro project with a custom image name, you must add `--platform=linux/amd64` to your command. For example:

Reviewer: ```suggestion
The Astro data plane is compatible only with AMD64-based Docker images and does not accept ARM64-based images. If you're using an Apple M1 computer to deploy an Astro project with a pre-built image name, you must add `--platform=linux/amd64` to your command. For example:
```

Do you add this to the command, or when you build the image? I think it's when you build the image in the first place? @wolfier can you confirm?

You can deploy a prebuilt Docker image to:

- Build a Docker image once and promote the Docker image across Deployments instead of rebuilding it each time.
- Reduce the time it takes to deploy. If your Astro project has a number of packages that take a long time to install, it can be more efficient to build it separately.
- Specify additional mounts and arguments in your project, which is required for setups such as [installing Python packages from private sources](develop-project.md#install-python-packages-from-private-sources).

To use this command, the image you deploy must be based on Astro Runtime and be available in a local Docker registry. 

You can use the `--image-name` flag with the following local development commands to test your image before deploying:

- `astro dev start`
- `astro dev restart`
- `astro dev parse`
- `astro dev pytest`

For more information about this command, see the [CLI command reference](cli/astro-deploy.md).

:::info

If you're using an Apple M1 computer to deploy an Astro project with a custom image name, you must add `--platform=linux/amd64` to your command. For example:

```sh

> astro deploy --image-name <custom-image> --platform=linux/amd64

Reviewer: ```suggestion
docker build --platform=linux/amd64
```

```

> 

Reviewer: ```suggestion

Ignore the warning that reads:

```
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
```
```
